source/AI.cpp

```diff
@@ -463,13 +463,14 @@ void AI::Step(const PlayerInfo &player)
 			
 			command |= AutoFire(*it);
 		}
-		if(isPresent && personality.Harvests() && DoHarvesting(*it, command))
+		if(isPresent && (personality.Harvests() || (it->IsYours() && Preferences::Has("Escorts harvest flotsam")))
+				&& DoHarvesting(*it, command))
 		{
 			it->SetCommands(command);
 			continue;
 		}
-		if(isPresent && personality.IsMining() && !target
-				&& it->Cargo().Free() >= 5 && ++miningTime[&*it] < 3600 && ++minerCount < 9)
+		if(isPresent && (personality.IsMining() || (it->IsYours() && Preferences::Has("Escorts target asteroids")))
+				&& !target && it->Cargo().Free() >= 5 && ++miningTime[&*it] < 3600 && ++minerCount < 9)
```

Comment:
Do you have any better idea? That seemed straight forward as it is consistent with the mining personality.
But thinking about your comment: If I understand you right, you mean orders like "hold position", "move to" ?
So basically you say that the regular commands like "hold" (agreed!), "move to location x" or "gather" (really?) should go first?